src/compiler/checker/lines_24000_32000.rs

```diff
@@ -20,6 +20,9 @@ impl TypeChecker {
         check_mode: Option<CheckMode>,
     ) -> Rc<Type> {
         let symbol = self.get_resolved_symbol(node);
+        if Rc::ptr_eq(&symbol, &self.unknown_symbol()) {
+            return self.error_type();
```

Comment:
First, when it tries to resolve the `Symbol` of an unknown variable reference that returns `unknownSymbol` (a singleton `Symbol` on the type checker instance), and then here when it's trying to get the type of the symbol it returns the singleton `Type` `errorType` from the type checker instance for an unknown symbol

`errorType` has the `TypeFlags.Any` flag set on it, and so that basically makes there be no further type errors raised from usage of the unknown variable (because `any` basically disables type-checking)